libyul/optimiser/KnowledgeBase.cpp

```diff
@@ -51,21 +50,42 @@ bool KnowledgeBase::knownToBeDifferent(YulString _a, YulString _b)
 	return false;
 }
 
+optional<u256> KnowledgeBase::differenceIsKnownConstant(YulString _a, YulString _b)
```

Comment:
```suggestion
optional<u256> KnowledgeBase::differenceIfKnownConstant(YulString _a, YulString _b)
```
Otherwise it sounds like it returns a ``bool``.